wandb/internal/internal.py

```diff
@@ -33,7 +35,30 @@
 logger = logging.getLogger(__name__)
 
 
+_exited = False
+
+
+@atexit.register
+def handle_exit(*args):
+    global _exited
+    if not _exited:
+        _exited = True
+        exc_type, exc_value, exc_traceback = sys.exc_info()
+        if exc_traceback:
+            logger.exception("Internal process exited with exception:")
+        else:
+            logger.info("Process exited cleanly")
+
+
+# TODO: we may not need these, for now we may as well
+signal.signal(signal.SIGTERM, handle_exit)
+signal.signal(signal.SIGINT, handle_exit)
```

Comment:
I added this in the hopes of getting better debug info on catastrophic failure.

Reply:
CAREFUL.THIS BREAKS our control-c handling.  (at least i think installing SIGINT will)